sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java

```diff
@@ -694,11 +614,10 @@ private Read() {}
        */
       public <X> Bound<X> withSchema(Class<X> type) {
         return new Bound<>(
-            name,
-            filenamePrefix,
+            outputPrefix,
```

Comment:
Might want to consider AutoValue (I believe jkff already has that coded up). Right now this feels a bit error prone, especially with multiple string arguments that are easy to mix up